src/simulators/statevector/qubitvector.hpp

```diff
@@ -516,128 +684,120 @@ cvector_t QubitVector<statevector_t>::vector() const {
 // Utility
 //------------------------------------------------------------------------------
 
-template <class statevector_t>
-void QubitVector<statevector_t>::initialize() {
-  const int_t end = num_states_;    // end for k loop
-
-#pragma omp parallel for if (num_qubits_ > omp_threshold_ && omp_threads_ > 1) num_threads(omp_threads_)
-  for (int_t k = 0; k < end; ++k)
-    statevector_[k] = 0.0;
-
-  statevector_[0] = 1.;
-}
-
-template <class statevector_t>
-void QubitVector<statevector_t>::initialize(const cvector_t &statevec) {
-  if (num_states_ != statevec.size()) {
-    std::stringstream ss;
-    ss << "QubitVector<statevector_t>::initialize input vector is incorrect length (";
-    ss << num_states_ << "!=" << statevec.size() << ")";
-    throw std::runtime_error(ss.str());
-  }
-
-  const int_t end = num_states_;    // end for k loop
+template <class data_t>
+void QubitVector<data_t>::zero() {
+  const int_t end = size_;    // end for k loop
 
 #pragma omp parallel for if (num_qubits_ > omp_threshold_ && omp_threads_ > 1) num_threads(omp_threads_)
-  for (int_t k = 0; k < end; ++k)
-    statevector_[k] = statevec[k];
-}
-
-template <class statevector_t>
-void QubitVector<statevector_t>::initialize(const statevector_t &statevec, const size_t num_states) {
-  if (num_states_ != num_states) {
-    std::stringstream ss;
-    ss << "QubitVector<statevector_t>::initialize input vector is incorrect length (";
-    ss << num_states_ << "!=" << num_states << ")";
-    throw std::runtime_error(ss.str());
+  for (int_t k = 0; k < end; ++k) {
+    data_[k] = 0.0;
```

Comment:
And here, I'd put a `std::fill(data_, data_+END, 0.0);`, and remove the OpenMP directive on top, as this is going to be optimized to some sort of `memset` operation, which is pretty fast.

Reply:
These were done this way so `data_` could be replaced with any structure that has an operator access method and doesn't necessarily work with the STL